### ISSUE TITLE
Accessibility "Links have a discernible name" audit fails to find ambiguous link text.

### ISSUE THREAD

**Issue by joelhsmith** [GoogleChrome/lighthouse#6737]
#### Provide the steps to reproduce
1. Run LH on https://joelhsmith.github.io/debug-examples/lh-tests/index.html

#### What is the current behavior?

When running Lighthouse 3.x, 4.x, or in the Chrome GUI Lighthouse or aXe fails to find links without "discernible name".  Notice it does not find "read more", "learn more", "here" in the example page. I think this is a regression.

https://github.com/joelhsmith/debug-examples/blob/master/lh-tests/read-more-prob-lh4.json

#### What is the expected behavior?

Identify "read more", "learn more", "here" as having bad link text and report under "link-name" as failing items.

#### Environment Information
* Affected Channels: CLI & DevTools?
* Lighthouse version: 
  * 4.0.0-alpha.2-3.2.1
  * 3.2.1
  * Chrome's Lighthouse GUI
* Node.js version: 8.9.1
* Operating System: OSX 10.12.6

**Related issues**

**Comment by patrickhulce** [comment]
Thanks for filing @joelhsmith! I'm not seeing that behavior on real v4 (v4.0.0-alpha.2-3.2.1 is really just 3.2.1 republished) or on v3.0.3.

![image](https://user-images.githubusercontent.com/2301202/49608559-8fd2d080-f95e-11e8-8bed-b4c80c396562.png)

(FWIW 'Read more' has never been flagged, just "Learn more" and "Here" 😃)

**Comment by joelhsmith** [comment]
It looks like your screenshot found it via the "Best Practices" category.  It should also show up under Accessibility as a different failing audit also. If you rerun with just "Accessibility", does it replicate my issue?

Co-workers can also replicate that it is not showing up under "Accessibility" category.  They are all on the same version of Chrome for Mac. Same result when testing with an incognito window.

Thanks for the tip RE: Not flagging "read more".  Is there any reason for that?

**Comment by patrickhulce** [comment]
Oh, I think you're confusing the purpose of the audit then. If you're talking about the `link-name` audit, it's never been trying to judge the content of the link text. It's just making sure it's unique and accessible to a screen reader.

https://dequeuniversity.com/rules/axe/2.2/link-name?application=lighthouse

**Comment by joelhsmith** [comment]
Thanks @patrickhulce !  aXe-core did not mark them as violations either.  I should have checked in in aXe before I created an issue.  So, this is works-as-designed. 

I updated the page to have 5 matching "read more" links and retested it, hoping that would trigger them as failing items in aXe-core.  But it did not.  Maybe I'll pursue it up-stream on the aXe issue queue to find out the rational for not marking it, there is probably a reason. 

Do you think it would be worthwhile to add a note to the new issue guidelines, like "If capable, test issue it in the source project first." That way you get less dumb issues like this to deal with :-) LOL

**Comment by patrickhulce** [comment]
No worries @joelhsmith you're definitely not a problem issue filer!!

Yeah might be worth a note in the issue template "If filing an accessibility issue, please see if your issue is reproducible with axe-core first" interested in a PR? 😉

**Comment by joelhsmith** [comment]
@patrickhulce I'd be happy to make the edit and do a PR! 

Are Performance, Best Practices, SEO, and PWA fully home grown Google Lighthouse projects.  And they do not have an alternate source engine to check first, like Accessibility does with aXe-core?

**Comment by patrickhulce** [comment]
> Are Performance, Best Practices, SEO, and PWA fully home grown Google Lighthouse projects. And they do not have an alternate source engine to check first

Yeah for the most part. There are a couple exceptions, but axe/Accessibility is definitely the unique case here where checking upstream first would help a lot :)
